src/mempool/mempool.ts

```diff
@@ -547,6 +596,24 @@ export class MemoryMempool {
         minOps?: number
     ): Promise<UserOperationInfo[]> {
         const outstandingUserOperations = this.store.dumpOutstanding().slice()
+
+        // Sort userops before the execution
+        // Decide the order of the userops based on the sender and nonce
+        // If sender is the same, sort by nonce key
+        outstandingUserOperations.sort((a, b) => {
+            const aUserOp = deriveUserOperation(a.mempoolUserOperation);
+            const bUserOp = deriveUserOperation(b.mempoolUserOperation);
+
+            if (aUserOp.sender === bUserOp.sender) {
+                const [aNonceKey,] = getNonceKeyAndValue(aUserOp.nonce);
```

Comment:
Nope, this one is needed for "soft guarantee", from the lattice doc

> Or by having a “soft guarantee” for numerically ordering user operations with different nonce key (ie if we used the current timestamp as nonce key, it would be nice if the user operations are ordered by timestamp)

Reply:
oh right, fair enough, but after sorting by key we should then also make sure that for each key the nonce values are in order, no?